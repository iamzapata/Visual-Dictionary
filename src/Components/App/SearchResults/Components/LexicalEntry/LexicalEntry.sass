.LexicalEntry
  h4, h5
    display: inline-block

  .Dialects
    margin: 0 5px 0 0

  .LexicalCategory
    display: inline-block
    margin: 0 0 5px 5px

  .Pronunciations
    margin: 0 0 15px 0