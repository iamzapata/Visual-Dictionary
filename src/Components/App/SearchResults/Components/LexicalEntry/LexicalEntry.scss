body {
  counter-reset: wordSection;
}

.LexicalEntry {
  counter-reset: definitionSection;

  h4,
  h5 {
    display: inline-block;
  }

  h4.title {
    position: relative;
    margin-right: 5px;
  }

  > h4.title::after {
    counter-increment: wordSection;
    content: counter(wordSection);
    font-weight: bolder;
    position: absolute;
    font-size: 10px;
  }

  .Dialects {
    margin: 0 5px 0 0;
  }

  .LexicalCategory {
    display: inline-block;
    margin: 0 0 5px 5px;
  }

  .Pronunciations {
    margin: 0 0 15px 0;
  }

  .Sense::before {
    counter-increment: definitionSection;
    content: counter(definitionSection) ". ";
    font-weight: bolder;
  }
}
