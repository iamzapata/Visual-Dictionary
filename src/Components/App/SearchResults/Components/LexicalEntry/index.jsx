import React from "react"
import { arrayOf, shape, string } from "prop-types"
import PlayAudioButton from "../PlayAudioButton"
import "./LexicalEntry.sass"

const renderPronunciations = pronunciations => (
  <div>
    {pronunciations.map(p => (
      <span key={p.id}>
        /<span>{p.phoneticSpelling}</span>/
        <p>
          {
            <span className="Dialects">
              {p.dialects.map(d => <span key={d}>{d}</span>)}
            </span>
          }
          {p.audioFile && <PlayAudioButton audioFile={p.audioFile} />}
        </p>
      </span>
    ))}
  </div>
)

const LexicalEntry = ({ entry }) => {
  return (
    <div className="LexicalEntry">
      <h4 className="title is-4">{entry.text}</h4>
      <h7 className="is-italic">{entry.lexicalCategory}</h7>
      {renderPronunciations(entry.pronunciations)}
    </div>
  )
}

LexicalEntry.propTypes = {
  entry: shape({
    entries: arrayOf(shape({})),
    language: string,
    lexicalCategory: string,
    pronunciations: arrayOf(shape({})),
    text: string
  }).isRequired
}

export default LexicalEntry
