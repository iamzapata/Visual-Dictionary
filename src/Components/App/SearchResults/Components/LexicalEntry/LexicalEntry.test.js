import React from "react"
import LexicalEntry from "./index.jsx"
import { render, cleanup } from "react-testing-library"
import mockEntry from "../../mocks.js"

describe("<LexicalEntry />", () => {
  afterEach(cleanup)

  const createProps = () => ({
    entry: {
      ...mockEntry()
    }
  })

  it("Should render self", () => {
    const { container } = render(<LexicalEntry {...createProps()} />)

    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("Should have a LexicalEntry className", () => {
    const { container } = render(<LexicalEntry {...createProps()} />)
    const classNameSelection = container.querySelectorAll(".LexicalEntry")

    expect(classNameSelection.length).toEqual(1)
  })

  it("Should display the entry's text", () => {
    const { queryByText } = render(<LexicalEntry {...createProps()} />)

    const entryText = createProps().entry.text

    const entryTextPresence = queryByText(entryText)

    expect(entryTextPresence).toBeTruthy()
  })

  it("Should display lexical category", () => {
    const { queryByText } = render(<LexicalEntry {...createProps()} />)

    const lexicalCategory = createProps().entry.lexicalCategory

    const lexicalCategoryPresence = queryByText(lexicalCategory)

    expect(lexicalCategoryPresence).toBeTruthy()
  })

  it("Should display etymologies", () => {
    const { queryByText } = render(<LexicalEntry {...createProps()} />)
  })
})
