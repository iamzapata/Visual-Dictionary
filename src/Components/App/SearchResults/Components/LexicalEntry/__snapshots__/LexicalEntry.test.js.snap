// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LexicalEntry /> Should render self 1`] = `
<div
  class="LexicalEntry"
>
  <h4
    class="title is-4"
  >
    privacy
  </h4>
  <h5>
    Noun
  </h5>
  <div>
    <span>
      /
      <span>
        prɪvəsi
      </span>
      /
      <p>
        <span
          class="Dialects"
        >
          <span>
            British English
          </span>
        </span>
        <button
          class="PlayAudioButton"
        >
          ►
        </button>
      </p>
    </span>
    <span>
      /
      <span>
        prʌɪvəsi
      </span>
      /
      <p>
        <span
          class="Dialects"
        >
          <span>
            British English
          </span>
        </span>
      </p>
    </span>
  </div>
</div>
`;
